### ISSUE TITLE
How to reschedule a card via AnkiDroidJS?

### ISSUE THREAD

**Issue by MagTun** [ankidroid/Anki-Android#9727]
Is there a way to reschedule a card via AnkiDroidJS? By reschedule I mean preforming the same action that is done when using the button "reschedule for review in x days".

**Is your feature request related to a problem? Please describe.**
I have written a script  in my card template that determine if a card has an interval above 30 days and if yes, will first show either the front or the back based on the evenness of the number of review for this card. 
If I see the front first, then I will dispose the card as normal (with one of the 4 the Anki buttons: again,...).
But if I see the back first, I want to add a HTML button to reschedule the card in X days (X been the number of days of the current interval). I don't want to use the Anki button so that the current interval and other settings of the card aren't updated by this review. 

Is this currently possible?

**Comment by WiliTest** [comment]
Is it still in progress? This function is really essential, please don't close it.

**Comment by krmanik** [comment]
It is in development.

**Comment by MagTun** [comment]
Amazing! Thanks a lot for your work!

**Comment by MagTun** [comment]
Thanks for adding this new feature!

I have just tested the function `AnkiDroidJS.ankiSetCardDue()` and it works great for rescheduling the cards but there is a problem: it changes the interval of the card to the one set in the function. 

But when I opened this issue, I was looking for a way to reschedule my cards without changing anything in their settings. 

The [anki doc](https://docs.ankiweb.net/browsing.html?highlight=reschedul#cards) _seems_ to say that it's possible if a `'!' is included at the end of the range`. But when I did that, there was an error `This card uses unsupported Ankidroid features. Contact developer [...] (Error code:7)`.

Any idea if rescheduling a card without changing its interval could be possible?

**Comment by krmanik** [comment]
I think in AnkiDroid the `!` by another option reset. So, it may be needs to included in the editbox. But for settings I don't think there are any changes after card get rescheduled except for last row.

|.|Ank| AnkiDroid|
|:--:|:--:|:--:|
|Rescheduling option|![image](https://user-images.githubusercontent.com/12841290/165789039-a95328e5-8cb8-4cd2-b48e-babb7182a515.png)|![image](https://user-images.githubusercontent.com/12841290/165790122-8ce3a9d0-385e-4179-80e3-e6568c739d90.png)|
|After rescheduled| ![image](https://user-images.githubusercontent.com/12841290/165789688-048ad357-5e5b-4588-9bad-bb2c295d182a.png) |![image](https://user-images.githubusercontent.com/12841290/165790028-8a7869cc-50e0-4736-958c-61158f97dc51.png)|

**Comment by MagTun** [comment]
Thanks for that. 
Does this mean that there is no other way to simply delay a card that is currently reviewed, to a couple of days, without changing anything including its interval? I know there is the "bury" option but it only delays by a day. So that would be a "bury but for 5 days". Is there something like that?

**Comment by krmanik** [comment]
I can confirm that AnkiDroid set the `Ease` to 250% and reset interval on changing due but Anki does not.
This is Anki Ecosystem related issues.

**Comment by mikehardy** [comment]
Can we be very very specific about which version of desktop and which version of AnkiDroid are in use in the comparison tests?
Then based on those versions we need to see where the inconsistency comes in, if it is something we are actively doing (in our java/kotlin libanki code here), or if it is just that we need to adopt a newer version of the underlying scheduler (rust lib in Anki-Android-Backend) or similar

**Comment by krmanik** [comment]
In AnkiDroid in this function at line 166
https://github.com/ankidroid/Anki-Android/blob/main/AnkiDroid/src/main/java/com/ichi2/anki/servicelayer/SchedulerService.kt#L162-L171

|.|Anki|AnkiDroid|
|:--:|:--:|:--:|
|Version|⁨2.1.50 (26d40c3a)|2.16alpha62|
|Before reschedule|![image](https://user-images.githubusercontent.com/12841290/165830685-d9461108-acdc-4829-910b-e71d61448cb4.png)|![image](https://user-images.githubusercontent.com/12841290/165831009-94b13988-1eae-456d-a85b-3c15266089df.png)|
|After reschedule|![image](https://user-images.githubusercontent.com/12841290/165830841-3d2845a7-d0a8-490f-a64c-d32d967261c3.png)|![image](https://user-images.githubusercontent.com/12841290/165831140-8275f830-b1c2-4538-9d81-391d88490be7.png)|⁩

**Comment by MagTun** [comment]
Same here on Anki Desktop 2.1.49 (dc80804a). When using "Set Due Date", the `interval` and ⁩` ease` doesn't change. The only setting that is change is the date for "Latest review":

Before 
![image](https://user-images.githubusercontent.com/15843700/165944437-5a5c0c87-38e6-47e2-a975-e248553cd773.png)

After 
![image](https://user-images.githubusercontent.com/15843700/165944442-0bd0fda2-680a-4c35-9175-422af95b1574.png)

**Comment by david-allison** [comment]
See also: https://github.com/ankidroid/Anki-Android/issues/9740

https://github.com/ankitects/anki/releases/tag/2.1.41

**Comment by krmanik** [comment]
It is duplicate. It should be closed because solving the #9740 issue will solve also this.

**Comment by MagTun** [comment]
The problem is not so much that the ease is reset to 250% (which is the problem mentioned in #9740) but that the interval is set to the one used for the rescheduling (if my cards' interval are 3 months, and I set a reschedule for these cards in 4 days, expecting the intervals to still be 3 months, I will find myself reviewing a lot of cards just to get back to the interval where they are currently).

Solving #9740 will fix the ease but it won't fix the problem with the interval.  

I understand that this request was closed because now there is a way to reschedule with AnkidroidJS  (which was the request here). But what about the interval: should I open a new issue for the interval or should I simply comment on #9740?
